src/mod_zone_difficulty_scripts.cpp

```diff
@@ -537,6 +537,89 @@ class mod_zone_difficulty_worldscript : public WorldScript
         sZoneDifficulty->IsDebugInfoEnabled = sConfigMgr->GetOption<bool>("ModZoneDifficulty.DebugInfo", false);
         sZoneDifficulty->LoadMapDifficultySettings();
     }
+
+    void OnAfterUpdateEncounterState(Map* map, EncounterCreditType type, uint32 creditEntry, Unit* source, Difficulty difficulty_fixed, DungeonEncounterList const* encounters, uint32 dungeonCompleted, bool updated)
+    {
+        LOG_ERROR("sql.sql", "Encounter completed");
+        // todo: Apparently this doesn't fire.
+        // Must set CompletedEncounterOnNormal to true, if the encounter wasn't in hardmode.
+    }
```

Comment:
This is only used if UpdateEncounterState() is passed.

You can see how it is used here (The only place in the core where it will update, hence why it doesnt work for you)
```
server\game\Spells\Spell.cpp	4490	        m_caster->FindMap()->UpdateEncounterState(ENCOUNTER_CREDIT_CAST_SPELL, m_spellInfo->Id, m_caster);
server\scripts\EasternKingdoms\BlackrockMountain\BlackrockDepths\blackrock_depths.cpp	-	1 match in 714 lines.
server\scripts\EasternKingdoms\BlackrockMountain\BlackrockDepths\blackrock_depths.cpp	271	                        me->GetMap()->UpdateEncounterState(ENCOUNTER_CREDIT_KILL_CREATURE, NPC_GRIMSTONE, me);
server\scripts\EasternKingdoms\BlackrockMountain\MoltenCore\boss_majordomo_executus.cpp	-	1 match in 696 lines.
server\scripts\EasternKingdoms\BlackrockMountain\MoltenCore\boss_majordomo_executus.cpp	287	                    me->GetMap()->UpdateEncounterState(ENCOUNTER_CREDIT_KILL_CREATURE, me->GetEntry(), me);
server\scripts\EasternKingdoms\Karazhan\instance_karazhan.cpp	-	1 match in 460 lines.
server\scripts\EasternKingdoms\Karazhan\instance_karazhan.cpp	174	                        instance->UpdateEncounterState(ENCOUNTER_CREDIT_KILL_CREATURE, 16812, nullptr);
server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_argent_challenge.cpp	-	1 match in 856 lines.
server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_argent_challenge.cpp	154	                    me->GetMap()->UpdateEncounterState(ENCOUNTER_CREDIT_CAST_SPELL, 68574, me); // paletress' spell credits encounter, but shouldn't credit achievements
```

Reply:
Thanks for the pointers. It works now!